x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -238,36 +276,139 @@ const getTransforms = async (options: { transformId?: string }, callAsCurrentUse
   return await callAsCurrentUser('transform.getTransforms', options);
 };
 
+async function getIndexPatternId(
+  indexName: string,
+  savedObjectsClient: SavedObjectsClientContract
+) {
+  const response = await savedObjectsClient.find<IIndexPattern>({
+    type: 'index-pattern',
+    perPage: 10,
+    search: `"${indexName}"`,
+    searchFields: ['title'],
+    fields: ['title'],
+  });
+
+  const ip = response.saved_objects.find(
+    (obj) => obj.attributes.title.toLowerCase() === indexName.toLowerCase()
```

Comment:
I know too little about the inner workings of Kibana index patterns and if they are case-insensitive. Can you explain why the `toLowerCase` is necessary or legit?

Reply:
Just checked with the Kibana team, I think index patterns aren’t case sensitive because index resolution on es isn’t case sensitive. So I don't think anyone will realistically be able to create `INDEX_PATTERN` or `Index_Pattern` either on UI side or on the API side. So you're right we might not need it. I think it's a small check but it's extra cost so I will remove the `.toLowerCase`.

Reply:
Removed in https://github.com/elastic/kibana/pull/67922/commits/1269b1d3f026e13d9b854f6afba6f58aedf6f295